pkgs/applications/networking/p2p/fragments/default.nix

```diff
@@ -1,76 +1,85 @@
 { lib
 , stdenv
 , fetchFromGitLab
-, meson
-, vala
-, ninja
-, pkg-config
-, wrapGAppsHook
-, desktop-file-utils
 , appstream-glib
-, python3
+, cmake
+, dbus
+, desktop-file-utils
+, git
 , glib
-, gtk3
-, libhandy
-, libtransmission
-, libb64
-, libutp
-, miniupnpc
-, dht
-, libnatpmp
-, libevent
-, curl
+, gtk4
+, libadwaita
+, meson
+, ninja
 , openssl
-, zlib
+, pkg-config
+, rustPlatform
+, sqlite
+, transmission
+, wrapGAppsHook4
 }:
 
 stdenv.mkDerivation rec {
   pname = "fragments";
-  version = "1.5";
+  version = "2.0.1";
 
   src = fetchFromGitLab {
     domain = "gitlab.gnome.org";
     owner = "World";
     repo = "Fragments";
     rev = version;
-    sha256 = "0x1kafhlgyi65l4w67c24r8mpvasg3q3c4wlgnjc9sxvp6ki7xbn";
+    sha256 = "sha256-3/v+MK7Zzn2tnUOe5cW35I5nJ+EsMUaqzTr25EloJ/k=";
   };
 
   patches = [
-    # Fix dependency resolution
-    ./dependency-resolution.patch
+    ./remove_meson_postinstall.patch
   ];
 
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-Js1UOPIegmQ8rJSPRen1knv0ngoVZXgAHsMbsm0tLcM=";
+  };
+
+  dontUseCmakeConfigure = true;
+
   nativeBuildInputs = [
+    appstream-glib
+    cmake
```

Comment:
Are you sure `cmake` is needed? Meson supports it for finding dependencies but typically, `pkg-config` will be sufficient.

Reply:
No, it's not needed. I added `pkg-config` after `cmake` and assumed I needed it.